### ISSUE TITLE
Hapi v17 Compatibility

### ISSUE THREAD

**Issue by Y-LyN-10** [dwyl/hapi-auth-google#51]
Hapi v17 will be released very soon. It introduces a lot of breaking changes and *probably no current plugin will work anymore*. Unfortunately that's the case for `hapi-auth-google` as well and it's not possible to make it backwards-compatible (new hapi [plugin API docs](https://hapijs.com/api/17.0.0-rc9#plugins)).

I have experimented a little bit in my [fork](https://github.com/Y-LyN-10/hapi-auth-google) to make it work with the latest version of Hapi ("17.0.0-rc9", but nodejs >= v8.x.x is required as well). 

 I am wondering how are we going to proceed with this in future? Will there be two distinct releases? Should I create a new branch or hold it for now?

I would be glad if we can discuss everything regarding to v17 in this thread. Thanks.

**Comment by nelsonic** [comment]
@Y-LyN-10 indeed Hapi v17 has _fundamentally_ changed how Plugins work ...  😞 
We are waiting for the Release (_Upgrade Path_) Notes to be finalised `before` doing any work on this: https://github.com/hapijs/hapi/issues/3658 
But if you have already made a start at migrating against the "Release Candidate" that's awesome! 👍

**Comment by Y-LyN-10** [comment]
Yes, I am waiting for the final release too: just in case there are more breaking changes :smile: 
Currently I have a _working_ (read: needs improvement) version of this library with updated example and tests (passing, yay) with Hapi v17.0.0-rc9. I would be _hapi_ to contribute here! ^_^

**Comment by Y-LyN-10** [comment]
@nelsonic, a new branch is created: [hapi17-compatible](https://github.com/dwyl/hapi-auth-google/tree/hapi17-compatible). What should I do now? :thinking: 
<img src="https://user-images.githubusercontent.com/6561599/32551736-2a822db0-c49a-11e7-9c0b-08b9cc32ad61.png" data-canonical-src="https://user-images.githubusercontent.com/6561599/32551736-2a822db0-c49a-11e7-9c0b-08b9cc32ad61.png" width="450" height="180" />

**Comment by nelsonic** [comment]
If you create a PR based on this branch Codecov should pass. 👍

**Comment by Y-LyN-10** [comment]
Just to clarify: PR to merge hapi17-compatible branch into ... master? :innocent:

**Comment by nelsonic** [comment]
@Y-LyN-10 Yeah, the PR should be into `master` thanks!
